src/mempool/mempool.rs

```diff
@@ -27,11 +27,15 @@ use tracing::{error, warn};
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Config {
     pub queue_size: u32,
+    pub allow_unlimited_mempool_size: bool,
 }
 
 impl Default for Config {
     fn default() -> Self {
-        Self { queue_size: 500 }
+        Self {
+            queue_size: 500,
+            allow_unlimited_mempool_size: false,
```

Comment:
Typically we just set queue_size to 0 to mean unlimited. But this is fine for now.